astropy/coordinates/earth.py

```diff
@@ -619,18 +644,42 @@ def get_gcrs_posvel(self, obstime):
         obsgeovel : `~astropy.coordinates.CartesianRepresentation`
             The GCRS velocity of the object
         """
-        # do this here to prevent a series of complicated circular imports
-        from .builtin_frames import GCRS
-
-        itrs = self.get_itrs(obstime)
-        geocentric_frame = GCRS(obstime=obstime)
         # GCRS position
-        obsgeoloc = itrs.transform_to(geocentric_frame).cartesian
-        vel_x = -OMEGA_EARTH * obsgeoloc.y
-        vel_y = OMEGA_EARTH * obsgeoloc.x
-        vel_z = 0. * vel_x.unit
-        obsgeovel = CartesianRepresentation(vel_x, vel_y, vel_z)
-        return obsgeoloc, obsgeovel
+        gcrs_data = self.get_gcrs(obstime).data
+        obsgeopos = gcrs_data.without_differentials()
+        obsgeovel = gcrs_data.differentials['s'].to_cartesian()
+        return obsgeopos, obsgeovel
+
+    def gravitational_redshift(self, obstime):
+        """Return the gravitational redshift at this EarthLocation.
+
+        Calculates the gravitational redshift, of order 3 m/s, due to the Sun,
+        Jupiter, the Moon, and the Earth itself.
+
+        Parameters
+        ----------
+        obstime : `~astropy.time.Time`
+            The ``obstime`` to calculate the redshift at.
+
+        Returns
+        --------
+        redshift :  `~astropy.units.Quantity`
+            Gravitational redshift in velocity units at given obstime.
+        """
+        # needs to be here to avoid circular imports
+        from .solar_system import get_body_barycentric
+        names = ('sun', 'jupiter', 'moon', 'earth')
```

Comment:
Perhaps this should be a keyword argument with this as the default instead of hard-coded? That way if say, you thought you really cared about `Saturn` for whatever reason, you could include it.

(If we go with my private->public scheme this could be done in the follow-on PR. )

Reply:
I think this is a great idea but will leave it to a follow-on PR to avoid holding this up, since one wants to be careful about documenting which names are OK, and matching masses with names...